lib/Settings/Admin.php

```diff
@@ -39,24 +40,31 @@ class Admin implements ISettings {
 	private $appName;
 
 	/** @var EventDispatcherInterface */
+	private $legacyEventDispatcher;
+	/** @var IEventDispatcher */
 	private $eventDispatcher;
 
 	/**
 	 * @param string $appName
 	 * @param IL10N $l
-	 * @param EventDispatcherInterface $eventDispatcher
+	 * @param EventDispatcherInterface $legacyEventDispatcher
 	 */
-	public function __construct($appName, IL10N $l, EventDispatcherInterface $eventDispatcher) {
+	public function __construct($appName, IL10N $l, EventDispatcherInterface $legacyEventDispatcher, IEventDispatcher $eventDispatcher) {
 		$this->appName = $appName;
 		$this->l10n = $l;
+		$this->legacyEventDispatcher = $legacyEventDispatcher;
 		$this->eventDispatcher = $eventDispatcher;
 	}
 
 	/**
 	 * @return TemplateResponse
 	 */
 	public function getForm() {
-		$this->eventDispatcher->dispatch('OCP\WorkflowEngine::loadAdditionalSettingScripts');
+		// @deprecated in 20.0.0: retire this one in favor of the typed event
+		$this->legacyEventDispatcher->dispatch('OCP\WorkflowEngine::loadAdditionalSettingScripts');
+		if (class_exists(LoadSettingsScriptsEvent::class)) {
```

Comment:
The app is stable-branched, so this check can be removed as this version is only working with 20+ anyway and there the event exists.

Reply:
You are right 👍 I missed that.